W3ChampionsIdentificationService/RolesAndPermissions/RolesAndPermissionsValidator.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using W3ChampionsIdentificationService.Middleware;
+using W3ChampionsIdentificationService.RolesAndPermissions.Contracts;
+
+namespace W3ChampionsIdentificationService.RolesAndPermissions
+{
+    public class RolesAndPermissionsValidator
+    {
+        private readonly IPermissionsRepository _permissionsRepository;
+        private readonly IRolesRepository _rolesRepository;
+        private readonly IUsersRepository _usersrepostiory;
+        public RolesAndPermissionsValidator(
+            IPermissionsRepository permissionsRepository, 
+            IRolesRepository rolesRepository, 
+            IUsersRepository usersRepository)
+        {
+            _permissionsRepository = permissionsRepository;
+            _rolesRepository = rolesRepository;
+            _usersrepostiory = usersRepository;
+        }
+
+        public void ValidatePermissionHttp(Permission permission)
+        {
+            if (string.IsNullOrEmpty(permission.Id))
+            {
+                throw new HttpException(400, "Id cannot be null or empty");
+            }
+
+            if (string.IsNullOrEmpty(permission.Description))
+            {
+                throw new HttpException(400, "Description cannot be null or empty");
+            }
+        }
+
+        public void ValidateRoleHttp(Role role)
+        {
+
+            if (string.IsNullOrEmpty(role.Id))
+            {
+                throw new HttpException(400, "Id cannot be null or empty");
+            }
+
+            if (string.IsNullOrEmpty(role.Description))
+            {
+                throw new HttpException(400, "Description cannot be null or empty");
+            }
+        }
+
+        public async Task ValidateUserDtoHttp(UserDTO user, bool isCreating = true)

```

Comment:
I think you don't need the parameter isCreating. You can check is user.Id has value. If it has value then it's not creating